handler-proxy/src/main/java/io/netty/handler/proxy/ProxyHandler.java

```diff
@@ -287,7 +287,7 @@ private void setConnectSuccess() {
             connectTimeoutFuture.cancel(false);
         }
 
-        if (connectPromise.trySuccess(ctx.channel())) {
+        if (!connectPromise.isDone()) {
```

Comment:
I guess an alternative solution would be to move `finished = true;` statement right after we remove both codecs (so that the subsequent `write` triggered in the listener will be enqueued). I honestly can't tell the difference between these two approaches so LGTM!